### PR TITLE
feat(aws-lambda): add the amazon connect contact flow trigger

### PR DESCRIPTION
**Pull request by zach-anthony**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <>
- [ ]  If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include tests for your changes
- [ ]  If you are making substantial changes, consider adding a tslint.json containing { "extends": "dtslint/dt.json" }. If for reason the any rule need to be disabled, disable it for that line using // tslint:disable-next-line [ruleName] and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/aws-lambda/index.d.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@
 //                 Alessandro Palumbo <https://github.com/apalumbo>
 //                 Sachin Shekhar <https://github.com/SachinShekhar>
 //                 Ivan Martos <https://github.com/ivanmartos>
+//                 Zach Anthony <https://github.com/zach-anthony>
```
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts:39]: Just to check you're OK with getting contacted for every PR on this package?

File: types/aws-lambda/trigger/connect-contact-flow.d.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            InstanceARN: string;
+            PreviousContactId: string;
+            Queue: string | null;
+            SystemEndpoint: ConnectContactFlowEndpoint | null;
+            MediaStreams: {
+                Customer: {
+                    Audio: {
+                        StartFragmentNumber?: string;
+                        StartTimestamp?: string;
+                        StreamARN?: string;
```
Comment (simonbuchan) [inline types/aws-lambda/trigger/connect-contact-flow.d.ts:26]: Just wondering if these are definitely sometimes not present, while the above are instead sometimes `null`? AWS is oftew pretty weird about these things.

**Review by simonbuchan — APPROVED** [review]
Looks great to me!

File: types/aws-lambda/index.d.ts (hunk 1 above)
Comment (zach-anthony) [reply types/aws-lambda/index.d.ts:39]: Yeah, that's fine

File: types/aws-lambda/trigger/connect-contact-flow.d.ts (hunk 2 above)
Comment (zach-anthony) [reply types/aws-lambda/trigger/connect-contact-flow.d.ts:26]: Yeah, the AWS documentation is pretty poor for Amazon Connect Lambda triggers, but from all my testing those attributes are definitely sometimes not present, but the other ones are present with null values

Comment (simonbuchan) [reply types/aws-lambda/trigger/connect-contact-flow.d.ts:26]: Yeah, pretty standard stuff from them in both counts. Thanks for verifying this!

File: types/aws-lambda/tsconfig.json [hunk 3]
```diff
@@ -26,6 +26,7 @@
         "test/cloudwatch-tests.ts",
         "test/codepipeline-tests.ts",
         "test/codebuild-tests.ts",
+        "test/connect-contact-flow-tests.ts",
```
Comment (simonbuchan) [inline types/aws-lambda/tsconfig.json:29]: Hmm that's annoying, it seems adding tests triggers the config changed label so you don't get the nice new automated merge flow. They should probably fix that....

**Pull request merged**